ingestion/functions/parsing/cuba/cuba.py

```diff
@@ -16,6 +20,59 @@
             os.pardir,os.pardir, 'common'))
     import parsing_lib
 
+_PROVINCES = {}
+_MUNICIPALITIES = {}
+province_set = set()
+
+
+def _drop_admin2(municipality_location):
+    "Drops admin2 info and uses same data for province location"
+    newloc = copy.deepcopy(municipality_location)
+    newloc['name'] = newloc['administrativeAreaLevel1']
+    del newloc['administrativeAreaLevel2']
+    newloc['geoResolution'] = "Admin1"
+    return newloc
+
+# Geocoding data is from Wikipedia:
+# Municipalities of Cuba,
+# https://en.wikipedia.org/w/index.php?title=Municipalities_of_Cuba&oldid=1010874027
+# (version as of 21 August 2021)
+
+with (Path(__file__).parent / 'geocodes.csv').open() as geof:
+    reader = csv.DictReader(geof)
```

Comment:
I'd also wrap this in a `try/except` and log any errors

Reply:
Geocoding can't fail without the rest of it working, so I'd rather have it crash immediately. AWS Cloudwatch captures exception tracebacks.